src/core/plugin/luapi_application.h

```diff
@@ -1113,6 +1115,24 @@ static int applib_changeBackgroundPdfPageNr(lua_State* L) {
     return 1;
 }
 
+/*
+ * pushes an rectangle (width, height, x, y) to the lua stack as a new table
+ * in most cases you'll want to do a
+ * lua_newtable(L);  // create rectangle table
+ * before and a
+ * lua_setfield(L, -2, "snappedBounds");  // add rectangle table to other table
```

Comment:
Instead of "snappedBounds" the name of the used rectangle should be used.

Maybe factor the name into the `pushRectangleHelper` method and call `lua_newtable(L)` at the beginning and `lua_setfield(L, -2, name)` at the end of it. That would simplify things, although it could be argued that someone might want to add more info to a rectangle in future, which would make it impossible to use this helper function. What do you think?

Reply:
> it could be argued that someone might want to add more info to a rectangle in future, which would make it impossible to use this helper function

I've been exactly at this point when writing that function and this was the reason why I decided to really just pushing the corresponding values while noting that in most cases these two lines will surround the function-call.
So for being able to just extend a table I'd keep it this way. (That's my take on this design decision)

Reply:
I'm fine with this decision.